File: src/IsbInterop.CSharp/References/Wrappers/CompRecordInfo.cs
﻿using IsbInterop.Base.Wrappers;

namespace IsbInterop.References.Wrappers
{
  /// <summary>
  /// Обертка над ICompRecordInfo.
  /// </summary>
  internal abstract class CompRecordInfo : ObjectInfo, ICompRecordInfo
  {
    /// <summary>
    /// Код записи.
    /// </summary>
    public string Code
    {
      get { return (string)this.GetRcwProperty("Code"); }
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwICompRecordInfo">COM-объект ICompRecordInfo.</param>
    /// <param name="scope">Область видимости.</param>
    protected CompRecordInfo(object rcwICompRecordInfo, IScope scope) : base(rcwICompRecordInfo, scope) { }
  }
}

File: src/IsbInterop/References/Wrappers/CompRecordInfo.vb
Imports IsbInterop.Base.Wrappers

Namespace References.Wrappers
    ''' <summary>
    ''' Обертка над ICompRecordInfo.
    ''' </summary>
    Friend MustInherit Class CompRecordInfo
        Inherits ObjectInfo
        Implements ICompRecordInfo

        ''' <summary>
        ''' Код записи.
        ''' </summary>
        Public ReadOnly Property Code() As String Implements ICompRecordInfo.Code
            Get
                Return DirectCast(Me.GetRcwProperty("Code"), String)
            End Get
        End Property

        ''' <summary>
        ''' Конструктор.
        ''' </summary>
        ''' <param name="rcwICompRecordInfo">COM-объект ICompRecordInfo.</param>
        Protected Sub New(rcwICompRecordInfo As Object)
            MyBase.New(rcwICompRecordInfo)
        End Sub
    End Class
End Namespace